src/ContextualPopup/ContextualPopup.js

```diff
@@ -538,19 +556,41 @@ module.exports = kind(
 	/**
 	* @private
 	*/
-	showHideScrim: function (inShow) {
-		if (this.floating && (this.scrim || (this.modal && this.scrimWhenModal))) {
-			var scrim = this.getScrim();
-			if (inShow && this.modal && this.scrimWhenModal) {
-				// move scrim to just under the popup to obscure rest of screen
-				var i = this.getScrimZIndex();
-				this._scrimZ = i;
-				scrim.showAtZIndex(i);
-			} else {
-				scrim.hideAtZIndex(this._scrimZ);
-			}
-			util.call(scrim, 'addRemoveClass', [this.scrimClassName, scrim.showing]);
+	updateScrim: function (old, value, source) {
+		if (this._updateScrimMutex) return;
+		this._updateScrimMutex = true;
+
+		// We sync modal and spotlightModal here because binding doesn't guarantee sequence
+		// when it is used with observers.
+		if (source == 'modal') this.set('spotlightModal', this.modal);
+		if (source == 'spotlightModal') this.set('modal', this.spotlightModal);
+		this.showHideScrim(this.showing);
+
+		this._updateScrimMutex = false;
+	},
+
+	/**
+	* @private
+	*/
+	showHideScrim: function (show) {
+		var scrim = this.getScrim();
+
+		if (this._scrim && scrim != this._scrim) {
+			// hide if there was different kind of scrim
+			this._scrim.hideAtZIndex(this._scrimZ);
+			util.call(this._scrim, 'addRemoveClass', [this.scrimClassName, this._scrim.showing]);
+		}
+
+		if (show && this.floating && (this.scrim || (this.modal && this.scrimWhenModal))) {
+			// move scrim to just under the popup to obscure rest of screen
+			var i = this.getScrimZIndex();
+			this._scrimZ = i;
+			scrim.showAtZIndex(i);
+		} else {
+			scrim.hideAtZIndex(this._scrimZ);
 		}
+		util.call(scrim, 'addRemoveClass', [this.scrimClassName, scrim.showing]);
```

Comment:
And, I don't understand the intention of using util.call instead of below just like moon.Popup.
`scrim.addRemoveClass(this.scrimClassName, scrim.showing);`